### PR TITLE
Add outputting available updates

### PR DESCRIPTION
**Pull request by timja**
Fixes https://github.com/jenkinsci/plugin-installation-manager-tool/issues/99
Fixes https://github.com/jenkinsci/plugin-installation-manager-tool/issues/88

supports yaml, stdout, and txt

```
 timja at Frankel in ~/projects/jenkinsci/plugin-installation-manager-tool on add-cli-to-update*
$ java -jar plugin-management-cli/target/jenkins-plugin-manager-1.2.1-SNAPSHOT.jar --available-updates --plugins mailer:1.31
Available updates:
mailer (1.31) has an available update: 1.32

 timja at Frankel in ~/projects/jenkinsci/plugin-installation-manager-tool on add-cli-to-update*
$ java -jar plugin-management-cli/target/jenkins-plugin-manager-1.2.1-SNAPSHOT.jar --available-updates --output yaml --plugins mailer:1.31
plugins:
- artifactId: "mailer"
  source:
    version: "1.32"

 timja at Frankel in ~/projects/jenkinsci/plugin-installation-manager-tool on add-cli-to-update*
$ java -jar plugin-management-cli/target/jenkins-plugin-manager-1.2.1-SNAPSHOT.jar --available-updates --output txt --plugins mailer:1.31
mailer:1.32
 timja at Frankel in ~/projects/jenkinsci/plugin-installation-manager-tool on add-cli-to-update*
```

### CODE REVIEW

**Comment by nre-ableton** [comment]
I can't offer a good code review, since I don't know this codebase well enough, but I am definitely looking forward to an `--output` option. My use-case currently involves redirecting `stdout` and using a Python script to parse the output of the tool. Being able to directly parse JSON/YAML would be a great improvement!

**Comment by timja** [comment]
> I can't offer a good code review, since I don't know this codebase well enough, but I am definitely looking forward to an `--output` option. My use-case currently involves redirecting `stdout` and using a Python script to parse the output of the tool. Being able to directly parse JSON/YAML would be a great improvement!

this currently supports yaml, txt, and stdout. You shouldn't need to do any parsing at all

File: plugin-management-library/src/main/java/io/jenkins/tools/pluginmanager/parsers/YamlPluginOutputConverter.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        public PluginHolder(List<PluginYaml> plugins) {
+            this.plugins = plugins;
+        }
+
+        public List<PluginYaml> getPlugins() {
+            return plugins;
+        }
+    }
+
+    public static class PluginYaml {
```
Comment (jetersen) [inline plugin-management-library/src/main/java/io/jenkins/tools/pluginmanager/parsers/YamlPluginOutputConverter.java:50]: Please use the existing class? added in #114 ?

Comment (timja) [reply plugin-management-library/src/main/java/io/jenkins/tools/pluginmanager/parsers/YamlPluginOutputConverter.java:50]: 🤣 🙄 oops

Comment (jetersen) [reply plugin-management-library/src/main/java/io/jenkins/tools/pluginmanager/parsers/YamlPluginOutputConverter.java:50]: Feel free to rename them though... As naming is hard!

Comment (timja) [reply plugin-management-library/src/main/java/io/jenkins/tools/pluginmanager/parsers/YamlPluginOutputConverter.java]: refactored now, PTAL

File: plugin-management-library/src/main/java/io/jenkins/tools/pluginmanager/util/PluginListParser.java [hunk 2]
```diff
@@ -14,14 +19,15 @@
 import java.util.Arrays;
 import java.util.List;
 import org.apache.commons.lang3.StringUtils;
-import org.yaml.snakeyaml.Yaml;
```
Comment (jetersen) [inline plugin-management-library/src/main/java/io/jenkins/tools/pluginmanager/util/PluginListParser.java:17]: So we could remove snakeyamL?

Comment (timja) [reply plugin-management-library/src/main/java/io/jenkins/tools/pluginmanager/util/PluginListParser.java:17]: it's still used by jackson internally:
https://github.com/FasterXML/jackson-dataformats-text/tree/master/yaml

File: plugin-management-library/src/main/java/io/jenkins/tools/pluginmanager/impl/PluginManager.java [hunk 3]
```diff
@@ -349,23 +349,49 @@ public void showSpecificSecurityWarnings(List<Plugin> plugins) {
         }
     }
 
+    /**
+     * Takes a list of plugins and returns the latest version
+     * Returns existing version if no update
+     * @param plugins updated list of plugins
+     * @return latest plugin versions
+     */
+    public List<Plugin> getLatestVersionsOfPlugins(List<Plugin> plugins) {
```
Comment (slide) [inline plugin-management-library/src/main/java/io/jenkins/tools/pluginmanager/impl/PluginManager.java:358]: Is this supposed to be part of this PR as well? It's in the other one you submitted as well.

**Comment by slide** [comment]
One other thing, are the output YAML and TXT files supposed to be consumed by an invocation of this tool as well? If so, it might be good to have a round-trip type test.

File: plugin-management-library/src/main/java/io/jenkins/tools/pluginmanager/impl/PluginManager.java (hunk 3 above)
Comment (timja) [reply plugin-management-library/src/main/java/io/jenkins/tools/pluginmanager/impl/PluginManager.java:358]: it'll be removed from the diff once its merged

**Comment by timja** [comment]
> One other thing, are the output YAML and TXT files supposed to be consumed by an invocation of this tool as well? If so, it might be good to have a round-trip type test.

will take a look

File: plugin-management-library/src/main/java/io/jenkins/tools/pluginmanager/util/PluginListParser.java (hunk 2 above)
Comment (timja) [reply plugin-management-library/src/main/java/io/jenkins/tools/pluginmanager/util/PluginListParser.java:17]: removed the direct dep

**Pull request merged**